app/forms/legal_aid_applications/substantive_application_form.rb

```diff
@@ -0,0 +1,38 @@
+module LegalAidApplications
+  class SubstantiveApplicationForm
+    include BaseForm
+    form_for LegalAidApplication
+
+    after_validation :substantive_application_deadline_on
+
+    attr_accessor :substantive_application
+
+    validates :substantive_application, presence: { unless: :draft? }
+
+    def substantive_application_deadline_on
+      return model.substantive_application_deadline_on if model.substantive_application_deadline_on?
+
+      attributes['substantive_application_deadline_on'] ||= substantive_application_deadline
+    end
+
+    def save
+      super && model.provider_used_delegated_functions!
```

Comment:
I'm a bit uneasy by putting the change of state here because I think that for all the other states it's done in the controller.
So for consistency sake, it would be better to follow the same pattern.

Reply:
And it would mean the overloading of `save` would no longer be necessary. Nice one.